nexus_system_orchestrator/src/add_metadata.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2025 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <nexus_common/models/work_order.hpp>
+#include <yaml-cpp/yaml.h>
+
+#include "add_metadata.hpp"
+
+namespace nexus::system_orchestrator {
+
+BT::NodeStatus AddMetadata::tick()
+{
+  auto tasks = this->_ctx->get_tasks();
+
+  YAML::Node workcell_assignments = YAML::Load("[]");
+  for (std::size_t i = 0; i < tasks.size(); ++i)
+  {
+    const std::string& workcell_task_id = tasks.at(i).task_id;
```

Comment:
If  my previous comment is not valid and there is a need for two separate for loops I believe this can be refactored into a range based for loop since we only use `i` once and use it to access a vector, then we can get rid of the `tasks.at(i)` call.